### ISSUE TITLE
Support for initialization from archive NSData/NSStream

### ISSUE THREAD

**Issue by matt-curtis** [NSProgrammer/ZipUtilities#33]
Is this possible? We can do something like this with the current lib, but you have to save the NSData first as a file, which can be slow.

**Comment by NSProgrammer** [comment]
Just to clarify, you want support for ZIP archives that are not files, but either NSData of the entire archive or an NSStream of the entire archive?

**Comment by matt-curtis** [comment]
That's it exactly. More so NSData than NSStream, though.

**Comment by NSProgrammer** [comment]
Yeah, that's a good request.  I think the correct way to support that is to add abstractions for 1) a data source (can provide a file, NSData or stream) and 2) a data destination (can be a file or stream).

I am swamped by work, so if you want to fork and put in support on your own, that could unblock you.  Another option might be having your data be memory mapped to a file... just spitballing though :)

**Comment by matt-curtis** [comment]
Cool - I might just do that :) Currently this isn't blocking me, as having to use files isn't a problem yet. Should it, I'll definitely see if I can add it in.

**Comment by mickeyl** [comment]
Let me toss in another vote for this. I'm working with iOS 11 Drag'n'Drop and when dragging a DropBox folder from Files.app into my app, I get a ZIP of this folder in NSData format, hence it would be somewhat backwards having to save this first in a file before being able to use NOZ.
